docs/odcs.rst

```diff
@@ -0,0 +1,120 @@
+ODCS Integration
+================
+
+When a container build is started, the user usually provides the URL to a yum
+repo that contains the packages to be installed when building the container
+image. Usually, this is a puddle containing signed packages.
+
+In the case a yum repo URL is not provided, OSBS falls back to the yum repo
+associated with the build tag for the Koji build target (required parameter for
+container builds). Packages in this yum repo are not always signed. This is not
+ideal as it’s trivial to produce a container image that should never be shipped.
+
+Another problem with the current state is that to produce an image with signed
+packages, a parameter is needed (yum repo URL). To complicate matters, this yum
+repo URL will likely be different for future builds. This poses a huge problem
+for OSBS autorebuilds. In this case, OSBS will automatically rebuild layered
+images upon detected changes to parent image. It is necessary that these
+rebuilds pick up the latest, most relevant, signed packages.
+
+The On Demand Compose Service (`odcs`_) project has been created to address
+current limitations in obtaining a yum repo with signed packages. The general
+idea is to not require the user to provide a yum repo url when building
+container images.  Instead, OSBS will use ODCS to generate a yum repo with the
+relevant signed packages based on user configuration.
+
+API Changes
+-----------
+
+The following parameters will be used by ``buildContainer`` task defined by
+``koji-containerbuild`` plugin:
+
+``odcs_signed <bool, default True>`` When set to False, OSBS will not provide an
+empty list of ``sigkeys`` to ODCS causing the compose to potentially include
+unsigned packages. If ODCS integration is not in place, this parameter has no
+effect.
+
+``container.yaml`` Changes
+--------------------------
+
+In git repo, alongside Dockerfile, the ``container.yaml`` file will be enhanced
+to contain a "compose" section. This section will include all the required
+information for requesting a compose to be created by ODCS::
+
+    compose:
+      # Optional, default False.
+      inherit: true
+      # Optional, default "tag".
+      type: tag
+      # Required for "tag" type. Must contain at least one item.
+      packages:
+      - package-1-name # This is a package name, not an NVR.
+      - package-2-name
+      # Optional. Default value is configured in OSBS environment
+      sigkeys:
+      - AB123456
+
+Alternatively, for modules support::
+
+    compose:
+      type: module
+      # Eventually, ODCS will support ":" separator, for now "-" is used:
+      #     https://pagure.io/odcs/issue/98
+      # Multiple sources can be defined by separating them with a space.
+      # Required for "module" type.
+      source: "module_name1:stream1 module_name2:stream1"
```

Comment:
Should this be a string or a YAML list?

Reply:
I was keeping parity with how ODCS defines it. See bottom of page: https://pagure.io/odcs
Think it'll be cleaner to use something like `modules` here as a list of string?

Reply:
Changed it to `modules` list. I'm wondering if `type` now is necessary.